moveit_core/robot_state/test/robot_state_benchmark.cpp

```diff
@@ -66,312 +61,279 @@ Eigen::Isometry3d createTestIsometry()
-      benchmark::DoNotOptimize(result = isometry.inverse());
-      benchmark::ClobberMemory();
-    }
+    benchmark::DoNotOptimize(result = isometry.inverse());
+    benchmark::ClobberMemory();
   }
 }
 
 // Benchmark time to invert an Eigen::Affine3d(Eigen::Isometry).
 static void inverseAffineIsometry(benchmark::State& st)
 {
-  int n_iters = st.range(0);
-  Eigen::Isometry3d isometry = createTestIsometry();
-  Eigen::Affine3d affine;
-  affine.matrix() = isometry.matrix();
-
+  Eigen::Affine3d affine = createTestIsometry();
+  Eigen::Affine3d result;
   for (auto _ : st)
   {
-    for (int i = 0; i < n_iters; ++i)
-    {
-      Eigen::Affine3d result;
-      benchmark::DoNotOptimize(result = affine.inverse(Eigen::Isometry).affine());
-      benchmark::ClobberMemory();
-    }
+    benchmark::DoNotOptimize(result = affine.inverse(Eigen::Isometry));
+    benchmark::ClobberMemory();
   }
 }
 
 // Benchmark time to invert an Eigen::Affine3d.
 static void inverseAffine(benchmark::State& st)
 {
-  int n_iters = st.range(0);
-  Eigen::Isometry3d isometry = createTestIsometry();
-  Eigen::Affine3d affine;
-  affine.matrix() = isometry.matrix();
-
+  Eigen::Affine3d affine = createTestIsometry();
+  Eigen::Affine3d result;
   for (auto _ : st)
   {
-    for (int i = 0; i < n_iters; ++i)
-    {
-      Eigen::Affine3d result;
-      benchmark::DoNotOptimize(result = affine.inverse().affine());
-      benchmark::ClobberMemory();
-    }
+    benchmark::DoNotOptimize(result = affine.inverse());
+    benchmark::ClobberMemory();
   }
 }
 
 // Benchmark time to invert an Eigen::Matrix4d.
 static void inverseMatrix4d(benchmark::State& st)
 {
-  int n_iters = st.range(0);
-  Eigen::Isometry3d isometry = createTestIsometry();
-  Eigen::Affine3d affine;
-  affine.matrix() = isometry.matrix();
-
+  Eigen::Affine3d affine = createTestIsometry();
+  Eigen::Affine3d result;
   for (auto _ : st)
   {
-    for (int i = 0; i < n_iters; ++i)
-    {
-      Eigen::Affine3d result;
-      benchmark::DoNotOptimize(result = affine.matrix().inverse());
-      benchmark::ClobberMemory();
-    }
+    benchmark::DoNotOptimize(result = affine.matrix().inverse());
+    benchmark::ClobberMemory();
   }
 }
 
-// Benchmark time to construct a RobotState given a RobotModel.
-static void robotStateConstruct(benchmark::State& st)
+struct RobotStateBenchmark : ::benchmark::Fixture
 {
-  int n_states = st.range(0);
-  const moveit::core::RobotModelPtr& robot_model = moveit::core::loadTestingRobotModel(PANDA_TEST_ROBOT);
-
-  // Make sure the group exists, otherwise exit early with an error.
-  if (!robot_model->hasJointModelGroup(PANDA_TEST_GROUP))
+  void SetUp(const ::benchmark::State& /*state*/) override
   {
-    st.SkipWithError("The planning group doesn't exist.");
-    return;
-  }
-
-  for (auto _ : st)
-  {
-    for (int i = 0; i < n_states; i++)
+    if (rcutils_logging_set_logger_level("moveit_robot_model.robot_model", RCUTILS_LOG_SEVERITY_WARN) != RCUTILS_RET_OK)
```

Comment:
Looks like there's something missing inside this block?

Reply:
No. The if condition was just required to silent a compiler warning about unused function results. I don't care about the result.

Reply:
I see, then let's use `std::ignore` or a cast to `void` to express intent.
```suggestion
    std::ignore = rcutils_logging_set_logger_level("moveit_robot_model.robot_model", RCUTILS_LOG_SEVERITY_WARN);
```